Adds wizard item: Plague Talisman

EmeraldCandy opened the pull request:
<!-- By ticking or leaving ticked the option "Allow edits and access to secrets by maintainers" you give permission for repository maintainers to push changes to your branch without explicitly asking. -->

<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## What Does This PR Do

Adds the Plague Talisman to the wizard's spellbook for 2 points. This item is intended a side-grade equivalent to the necromatic stone. When purchased, the wizard receives 5 plague talismans. When used on a corpse with a mind in it, the victim will be turned into a (NON-NECRO  PLAGUE) zombie and is loyal to the wizard and makes wiz/zombies virus immune so they wont be affected by their own fun.. Each zombie is given a disease upon their creation that they carry, to which they are able to spread via attacks with the given claw items. The disease list has been curated for harsh but not biohazard level diseases, leaving out some nigh-gaurenteed death sentences such as GBS/Kuru. Most transformation diseases (besides advanced kingstons) have been omitted.

Currently, the zombie claws do 35 force, which is 5 less than the claymores given by the necromatic stone, due to both having more potential zombies and the additions of the plagues. This can be changed as needed. 

Additionally, three additional plague-zombie only diseases have been added:

_Mystic Malaise_
At maximum stages, will cause the user to vomit up "spacematter" vomit, which for 15 seconds will attempt to drain air out of a room before going inert.
Cure: Liquid dark matter

_Wand Rot_
At maximum stages, will slowly burn the user. With a low chance, users will sneeze, causing them to erupt a fireball out in a random direction.
Cure: Acetaldehyde

_Grut Gut_
At maximum stages, will sometimes cause the user to vomit with such intensity that they will be propelled backwards and crash into people/objects.
cure: Stabilizing Agent

## Why It's Good For The Game

New wizard items are interesting, and more options for minions could be fun. These thralls are similar to the necromatic stone, though technically weaker than the weapons given to the skeleton thralls. However, zombies are unable to wield weaponry, and rely on their claws, thus preventing the wizard from giving out equipment to their new thralls (such as staffs of resurrection).

## Images of changes

(thank you Exusa for the sprite!)
![dreammaker_5VG9m3nyfg](https://github.com/user-attachments/assets/b8e2b115-f3ef-453f-928d-773201a92eec)

## Testing

- Tested for converting alive/mindless/already slaved bodies
- ensured proper mindslaving
- ensured the victim gets the proper welcome message
- ensured the claws summon as intended
- ensured claws dispelled as intended
- ensured claws inflicted diseases as intended
- ensured each zombie was given their own individual disease
- ensured zombies rot properly
- ensured zombies revive properly
- ensured zombies could not be given items
- ensured mindless zombies dont spread the necro plague due to AI behavior
- ensured you couldnt manually spread zombie virus with a syringe
- infected self with each disease one at a time to ensure functionality

<hr>

### Declaration

- [X] I confirm that I either do not require [pre-approval](https://github.com/ParadiseSS13/Paradise/blob/master/docs/CODE_OF_CONDUCT.md#types-of-changes-that-need-approval) for this PR, or I have obtained such approval and have included a screenshot to demonstrate this below.
  <!-- Replace the box with [x] to mark as complete. -->
  <!-- Ensure there are no spaces between the x and the square brackets [] else this will not work properly. -->
<hr>

## Changelog

:cl:
add: Added plague talismans for the wizard
add: Added the plague claw spell for plague zombie use
add: Added three plague zombie diseases: Wand Rot, Grut Gut, Mystic Malaise
imageadd: Plague talisman icon
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->
<!-- If a PR has no impact on players (i.e. a code refactor that does not change functionality) then the entire Changelog heading and contents can be removed. -->

Scribble-Sheep commented on code/game/gamemodes/wizard/artefact.dm:
```diff
@@ -843,6 +843,82 @@ GLOBAL_LIST_EMPTY(multiverse)
+
+/obj/item/plague_talisman
+	name = "plague Talisman"
+	desc = "A vile rune capabale of raising the dead as plague-bearing creatures of destruction. The edges have sharp hooks"
```
```suggestion
	desc = "A vile rune, capable of raising the dead as plague-bearing creatures of destruction. The edges have sharp hooks."
```

Scribble-Sheep commented on code/game/gamemodes/wizard/artefact.dm:
```diff
@@ -843,6 +843,82 @@ GLOBAL_LIST_EMPTY(multiverse)
+//raises the victim into a special zombies and binds them to wiz
+/obj/item/plague_talisman/proc/raise_victim(mob/living/carbon/human/victim, mob/living/carbon/human/necromancer)
+
+	var/greet_text = "<span class='userdanger'>You have been raised into undeath by <b>[necromancer.real_name]</b>!\n[necromancer.p_theyre(TRUE)] your master now, assist them at all costs for you are now above death!</span>"
```
```suggestion
	var/greet_text = "<span class='userdanger'>You have been raised into undeath by <b>[necromancer.real_name]</b>!\n[necromancer.p_theyre(TRUE)] your master now, assist them at all costs, for you are now above death!</span>"
```

Scribble-Sheep commented on code/game/gamemodes/wizard/artefact.dm:
```diff
@@ -843,6 +843,82 @@ GLOBAL_LIST_EMPTY(multiverse)
+	var/static/list/plague_traits = list(TRAIT_LANGUAGE_LOCKED, TRAIT_ABSTRACT_HANDS, TRAIT_NOBREATH, TRAIT_I_WANT_BRAINS, TRAIT_NON_INFECTIOUS_ZOMBIE, TRAIT_PLAGUE_ZOMBIE)
+
+	victim.mind.add_antag_datum(new /datum/antagonist/mindslave/necromancy(necromancer.mind, greet_text))
+	victim.visible_message("<span class='warning'>[necromancer] places a vile rune onto the corpse's forehead where it adheres to the flesh. [victim]'s body rots and decays at unnatural speeds, before rising into a horrendous undead creature!</span>")
```
```suggestion
	victim.visible_message("<span class='warning'>[necromancer] places a vile rune upon [victim]'s lifeless forehead. The rune adheres to the flesh, and [victim]'s body rots and decays at unnatural speeds, before rising into a horrendous undead creature!</span>")
```

Scribble-Sheep commented on code/game/gamemodes/wizard/godhand.dm:
```diff
@@ -160,3 +160,11 @@
+
+/obj/item/melee/touch_attack/plague
+	name = "Plague and Pestilence"
+	desc = "Your hand glows dim with blisters and boils both real and imaginary. Yuck!"
```
```suggestion
	desc = "Your hand emits a dim glow, as blisters both real and imaginary sprout all over it. Yuck!"
```

Scribble-Sheep commented on code/game/gamemodes/wizard/spellbook.dm:
```diff
@@ -444,6 +444,13 @@
 
+/datum/spellbook_entry/item/plague_talisman
+	name = "Five Plague Talismans"
+	desc = "Special engraved runes for raising the dead into a plague-ridden animated corpse"
```
```suggestion
	desc = "Special engraved runes, used to raise the dead into plague-ridden, animated corpses."
```

Scribble-Sheep commented on code/game/objects/items/weapons/storage/belt.dm:
```diff
@@ -689,6 +689,22 @@
 
+/obj/item/storage/belt/plague_talisman
+	name = "Runic Belt"
+	desc = "A belt designed to hold safely contain plague ridden runes. Beter hope it doesnt start to drip."
```
```suggestion
	desc = "A belt designed to hold safely contain Plague Talismans. Better hope it doesn't start dripping..."
```

Scribble-Sheep commented on code/modules/antagonists/zombie/zombie_spells.dm:
```diff
@@ -154,3 +154,117 @@
+
+/obj/item/plague_claw
+	name = "Plague Claws"
+	desc = "Ichor-coated claws extending from your rotting hands, perfect for ripping bone and flesh for your master."
```
```suggestion
	desc = "Claws extending from your rotting hands, oozing a putrid ichor. Perfect for rending bone and flesh for your master."
```
This one's slightly less of a grammar touchup and more just reflavoring, feel free to reject if you prefer yur previous sentence.

EmeraldCandy replied:
resolved this one manually to fix another grammar error

Scribble-Sheep reviewed (commented):
Some grammar/spelling/flow touchups.

Scribble-Sheep commented:
And also, I see this has been reworked from being a direct virus vector into more of a conversion spell. I like it~
Hopefully it'll be fun in practice.

lewcc commented on code/game/gamemodes/wizard/artefact.dm:
```diff
@@ -843,6 +843,90 @@ GLOBAL_LIST_EMPTY(multiverse)
+	var/chosen_plague
+
+//checks if they're a valid target before trying to raise
+/obj/item/plague_talisman/attack__legacy__attackchain(mob/living/carbon/human/victim, mob/living/carbon/human/necromancer)
```
If you're adding new code, please make this use the new attack chain!

lewcc commented on code/game/gamemodes/wizard/artefact.dm:
```diff
@@ -843,6 +843,90 @@ GLOBAL_LIST_EMPTY(multiverse)
+	var/static/list/plague_traits = list(TRAIT_LANGUAGE_LOCKED, TRAIT_ABSTRACT_HANDS, TRAIT_NOBREATH, TRAIT_I_WANT_BRAINS, TRAIT_NON_INFECTIOUS_ZOMBIE, TRAIT_PLAGUE_ZOMBIE)
+
+	victim.mind.add_antag_datum(new /datum/antagonist/mindslave/necromancy(necromancer.mind, greet_text))
+	victim.visible_message("<span class='warning'>[necromancer] places a vile rune upon [victim]'s lifeless forehead. The rune adheres to the flesh, and [victim]'s body rots and decays at unnatural speeds, before rising into a horrendous undead creature!</span>")
```
```suggestion
	victim.visible_message("<span class='danger'>[necromancer] places a vile rune upon [victim]'s lifeless forehead. The rune adheres to the flesh, and [victim]'s body rots and decays at unnatural speeds, before rising into a horrendous undead creature!</span>")
```

lewcc commented on code/modules/antagonists/zombie/zombie_spells.dm:
```diff
@@ -154,3 +154,121 @@
+	// already have the disease, or have contracted it.
+	if(!target.HasDisease(claw_disease))
+		playsound(user.loc, 'sound/misc/moist_impact.ogg', 50, TRUE)
+		target.bleed_rate = max(5, target.bleed_rate + 1) // Them claws sharp! Good luck with whatever they were laced with
```
Might want to make them bleed regardless when they get hit?

EmeraldCandy replied:
changed

lewcc commented on code/modules/antagonists/zombie/zombie_spells.dm:
```diff
@@ -154,3 +154,121 @@
+		target.ContractDisease(plague)
+
+/obj/item/plague_claw/activate_self(mob/user)
+	if(..())
+		return
+	qdel(src)
```
What's this doing?

EmeraldCandy replied:
This is to let the claws be dropped either by "using" it or using the drop command. Theres not a big issue if they're accidently dispelled since they can be redeployed with no cooldown

lewcc commented on code/modules/mob/living/carbon/examine.dm:
```diff
@@ -306,6 +306,9 @@
 		msg += "[p_they(TRUE)] [p_are()] emitting a gentle blue glow!\n"
 
+	if(HAS_TRAIT(src, TRAIT_PLAGUE_ZOMBIE)) //to tell plague zombies easier through clothing
+		msg += "[p_they(TRUE)] smell like rot and death!\n"
```
This should make sure it shows up properly for he/she and they
"they smell like rot and death!"
"he smells like rot and death!"
```suggestion
		msg += "[p_they(TRUE)] smell[p_s()] like rot and death!\n"
```

EmeraldCandy commented on code/modules/antagonists/zombie/zombie_spells.dm:
```diff
@@ -154,3 +154,121 @@
+		qdel(claw)
+		return TRUE
+
+/datum/spell/plague_claws/can_cast(mob/user, charge_check, show_message)
+	var/mob/living/L = user
+	if(!L.get_active_hand() || istype(L.get_active_hand(), /obj/item/zombie_claw))
+		return ..()
```
misunderstood how this worked. refactored so its one-handed, and checks both hands to see if the claws are being held

EmeraldCandy commented on code/modules/antagonists/zombie/zombie_spells.dm:
```diff
@@ -154,3 +154,121 @@
+	if(dispel())
+		return
+	var/obj/item/plague_claw/claws = new /obj/item/plague_claw(user.loc, src, disease)
+	if(user.get_organ("l_hand") && user.get_organ("r_hand")) // Takes always takes both hand slots, but doesnt require both
```
the corpse must be dead first to raise, dropping anything that doesnt have antidrop and setting the hands to abstract hands only.  most antidrop is toggleable, however it was refactored regardless to only hold in one hand and check both hands for the spell

Contrabang commented on code/game/gamemodes/wizard/artefact.dm:
```diff
@@ -843,6 +843,89 @@ GLOBAL_LIST_EMPTY(multiverse)
+
+//raises the victim into a special zombies and binds them to wiz
+/obj/item/plague_talisman/proc/raise_victim(mob/living/carbon/human/victim, mob/living/carbon/human/necromancer)
+
+	chosen_plague = pick_disease()
+
+	var/datum/disease/plague = chosen_plague
+
+	var/greet_text = "<span class='userdanger'>You have been raised into undeath by <b>[necromancer.real_name]</b>!\n[necromancer.p_theyre(TRUE)] your master now, assist them at all costs, for you are now above death!<br> \
+		You have been bestowed the following plague: <br> \
+		[plague.name]!</span>"
+
+	var/static/list/plague_traits = list(TRAIT_LANGUAGE_LOCKED, TRAIT_ABSTRACT_HANDS, TRAIT_NOBREATH, TRAIT_I_WANT_BRAINS, TRAIT_NON_INFECTIOUS_ZOMBIE, TRAIT_PLAGUE_ZOMBIE)
+
+	victim.mind.add_antag_datum(new /datum/antagonist/mindslave/necromancy(necromancer.mind, greet_text))
+	victim.visible_message("<span class='danger'>[necromancer] places a vile rune upon [victim]'s lifeless forehead. The rune adheres to the flesh, and [victim]'s body rots and decays at unnatural speeds, before rising into a horrendous undead creature!</span>")
+
+	for(var/trait in plague_traits)
+		ADD_TRAIT(victim, trait, ZOMBIE_TRAIT)
+	victim.AddComponent(/datum/component/zombie_regen)
+	ADD_TRAIT(necromancer, TRAIT_VIRUSIMMUNE, MAGIC_TRAIT) // Cant have the user or zombie getting infected by their own plagues
+	ADD_TRAIT(victim, TRAIT_VIRUSIMMUNE, MAGIC_TRAIT)
+
+	var/datum/spell/plague_claws/plague_claws = new /datum/spell/plague_claws
+	victim.AddSpell(plague_claws)
+	plague_claws.disease = chosen_plague
+
+	//time to rot
+	if(!istype(victim))
+		return FALSE
+	for(var/obj/item/organ/limb as anything in victim.bodyparts)
+		if(!(limb.status & ORGAN_DEAD) && !limb.is_robotic())
+			limb.necrotize(TRUE, TRUE)
+
+	victim.heal_overall_damage(1000, 1000) // Cant very well have your new minions dead for so long if already hurt
```
Make them a non-infectious zombie, and give them a separate mindslave antag datum please. We should have duplicated code.

EmeraldCandy replied:
they already have the non_infectious_zombie trait. Created a new mindslave datum type based off the parent type necromancy.

EmeraldCandy replied:
completely reworked the antag type to run through the normal zombie process and created a new mindslave datum to use.

Contrabang commented on code/modules/antagonists/zombie/zombie_spells.dm:
```diff
@@ -154,3 +154,118 @@
+
+
+//wizard plague zombie claws
+/datum/spell/plague_claws
+	name = "Plague Claws"
+	desc = "Toggle your claws, allowing you to slash and infect people with deadly diseases."
+	action_icon_state = "vampire_claws"
+	action_background_icon_state = "bg_vampire"
+	human_req = TRUE
+	clothes_req = FALSE
+	antimagic_flags = NONE
+	base_cooldown = 0 SECONDS
+	var/list/our_claws = list()
+	var/disease
```
All of this and below is duplicate code also.

EmeraldCandy replied:
reduced duplicate code down by moving it into a child of zombie_claws

silverplatedelta commented:
Hey there, After some internal discuss, the balance team has some significant concerns about this item, with three members currently objecting to its addition in its current state. 

The three members objecting have left the following comments:

[Objector 1]:  major diseases can cripple a station running at normal capacity, add in a hectic round like wizard where doctors might not even stay in medbay and you have a 1/3 chance to instantly wipe out the station in one lucky cast of brainrot, gbs, etc. This would be way too strong.

[Objector 2]: Agreeing with [Objector 1] here, some of the diseses are ROUGH and all the wizard has to do is destoy medbay or viro to wipe the station quickly

[Objector 3] Agreeing with above, If the wizard gets rolling, this is going to kill or disable most of the crew.

Personally, I would suggest making it so the disease spread to players via claws cannot be otherwise spread (meaning if you get a flu zombie, that flu will only be spread via zombie attacks. ) This however would require ALOT more coding.

I would also suggest pinging the balance team in coding chat to discuss how to tune this item.

EmeraldCandy commented:
As suggested, made all diseases marked with a NON_CONTAGIOUS flag. Removed some diseases from the pick pool. Additionally, added three new wizard-only diseases: Grut Gut, Mystic Malaise, and Wand Rot

Stoniest commented:
Really like the new diseases and the fact that the cures for them aren't extremely simple and take a bit to make.

Contrabang commented on code/modules/antagonists/_common/antag_datum.dm:
```diff
@@ -311,7 +311,7 @@ GLOBAL_LIST_EMPTY(antagonists)
 	apply_innate_effects()
 	messages.Add(finalize_antag())
-	if(wiki_page_name)
+	if(wiki_page_name) // Prevent plague zombies double-box info
```
```suggestion
	if(wiki_page_name)
```

EmeraldCandy replied:
Resolved manually (github issues)

Contrabang commented on code/modules/antagonists/traitor/misc_mindslave.dm:
```diff
@@ -7,3 +7,10 @@
 	master_hud_name = "wizard"
+
+/datum/antagonist/mindslave/necromancy/plague_zombie
+
+	name = "Necromancy-risen Plague Zombie"
+	master_hud_name = "wizard"
+
```
```suggestion
	name = "Necromancy-risen Plague Zombie"
```

EmeraldCandy replied:
Resolved manually (github issues)

Contrabang commented on code/modules/antagonists/zombie/datum_zombie.dm:
```diff
@@ -11,21 +11,33 @@ RESTRICT_TYPE(/datum/antagonist/zombie)
 // possibly upgrades for the zombies after eating brains? Better vision (/datum/action/changeling/augmented_eyesight), better weapons (armblade), better infection, more inhereint armor (physiology)
 // ability to find nearby brains to eat (like cling/vamp ability to track people around them)
+/datum/antagonist/zombie/New(chosen_plague)
+	chosen_disease = chosen_plague
```
```suggestion
	chosen_disease = chosen_plague
	..()
```

EmeraldCandy replied:
Resolved manually (github issues)

Christer2222 commented:
Got this weird empty admin message with no text: 
![image](https://github.com/user-attachments/assets/60fc7d11-5990-4dff-83a2-96528fca5a85)

EmeraldCandy commented:
> Got this weird empty admin message with no text:

all fixed

lewcc commented on code/datums/diseases/zombie_virus.dm:
```diff
@@ -19,6 +19,15 @@
+	var/instant_zombie
+
+/datum/disease/zombie/New(chosen_plague, plague_zomb)
+	if(plague_zomb == TRUE)
```
```suggestion
	if(plague_zomb)
```

lewcc commented on code/game/gamemodes/wizard/artefact.dm:
```diff
@@ -842,6 +842,96 @@ GLOBAL_LIST_EMPTY(multiverse)
+	icon_state = "plague_talisman"
+	origin_tech = "bluespace=4;materials=4"
+	w_class = WEIGHT_CLASS_TINY
+	var/chosen_plague
+
+	new_attack_chain = TRUE
```
```suggestion
	new_attack_chain = TRUE
	var/chosen_plague
```

lewcc commented on code/game/gamemodes/wizard/artefact.dm:
```diff
@@ -842,6 +842,96 @@ GLOBAL_LIST_EMPTY(multiverse)
+		return
+
+	if(victim.mind.has_antag_datum(/datum/antagonist/mindslave/necromancy))
+		to_chat(necromancer, "<span class='warning'>This one has already been risen! Give it time.</span>")
```
```suggestion
		to_chat(necromancer, "<span class='warning'>This one is already under the artifact's influence! Give it time.</span>")
```

lewcc commented on code/game/gamemodes/wizard/artefact.dm:
```diff
@@ -842,6 +842,96 @@ GLOBAL_LIST_EMPTY(multiverse)
+
+	qdel(src) // talismans are single use
+
+	sleep(50) // so that mindslave comes after the zombie antag
+
+	var/greet_text = "<span class='userdanger'>You have been raised into undeath by <b>[necromancer.real_name]</b>!\n[necromancer.p_theyre(TRUE)] your master now, assist them at all costs, for you are now above death!<br> \
+		You have been bestowed the following plague: <br> \
+		[chosen_plague.name]!</span>"
+	victim.mind.add_antag_datum(new /datum/antagonist/mindslave/necromancy(necromancer.mind, greet_text, chosen_plague))
+
+	// Cant very well have your new minions dead for so long
+	victim.heal_overall_damage(1000, 1000)
+	victim.med_hud_set_health()
+	victim.med_hud_set_status()
+	victim.update_hands_hud()
+	victim.update_body()
+
```
Can you make this a timer, or does it need to be a sleep with the qdel(src)?

EmeraldCandy replied:
it has been made a timer

lewcc commented on code/game/gamemodes/wizard/artefact.dm:
```diff
@@ -842,6 +842,96 @@ GLOBAL_LIST_EMPTY(multiverse)
+//choose what disease this zombie will get
+/obj/item/plague_talisman/proc/pick_disease()
+	var/picked_disease
+	var/list/possible_diseases = list(
```
```suggestion
	var/static/list/possible_diseases = list(
```

lewcc commented on code/modules/antagonists/zombie/zombie_spells.dm:
```diff
@@ -29,14 +35,14 @@
 	var/mob/living/carbon/human/user = action.owner
 	var/obj/item/zombie_claw/claw = user.get_active_hand()
-	if(istype(claw, /obj/item/zombie_claw))
+	if(istype(claw, /obj/item/zombie_claw || /obj/item/zombie_claw/plague_claw))
```
I'm not sure this works how you'd expect it to, since the result of the boolean comp will result in `istype(claw, TRUE)`
```suggestion
	if(istype(claw, /obj/item/zombie_claw) || istype(claw, /obj/item/zombie_claw/plague_claw))
```

lewcc commented on code/modules/antagonists/zombie/zombie_spells.dm:
```diff
@@ -84,32 +94,37 @@
-	var/mob/living/carbon/human/target = atom_target
-	try_infect(target, user)
+	var/mob/living/carbon/human/attack_target = target
+	if(HAS_TRAIT(user, TRAIT_PLAGUE_ZOMBIE))
+		try_virus_infect(attack_target, user, claw_disease)
+		return
+	else
+		try_infect(attack_target, user)
```
unnecessary after the return
```suggestion
	if(HAS_TRAIT(user, TRAIT_PLAGUE_ZOMBIE))
		try_virus_infect(attack_target, user, claw_disease)
		return
	
	try_infect(attack_target, user)
```

1080pCat commented:
After one round of being a zomb

The plague claws should deal less damage to humans, they're a bit strong currently. Maybe each hit advancing the disease?

Also the wizard can understand zombies, but cant /speak/ zombie, leading to iffy one way communication charades

EmeraldCandy commented on code/datums/diseases/wizard_diseases.dm:
```diff
@@ -0,0 +1,232 @@
+	time_to_live = world.time + 15 SECONDS
+
+/obj/effect/decal/cleanable/vomit/spacematter/process()
+	var/datum/milla_safe/spacematter_vomit_process/milla = new()
```
the vomit now qdel's itself afterwards

EmeraldCandy commented on code/datums/diseases/wizard_diseases.dm:
```diff
@@ -0,0 +1,232 @@
+			if(prob(3))
+				to_chat(affected_mob, "<span class='danger'>You feel bloated.</span>")
+			if(prob(2))
+				to_chat(affected_mob, "<span class='danger'>Your feel an uncomfortable pressure in your abdomen</span>")
```
checked through for punctuation

EmeraldCandy commented:
> The plague claws should deal less damage to humans, they're a bit strong currently. Maybe each hit advancing the disease?

Moved claw damage down to 26 from 35 (4 hits will crit, down from 3) and made hits have a 40% chance to advance their respective disease

> Also the wizard can understand zombies, but cant /speak/ zombie, leading to iffy one way communication charades

Gave the wizard the ability to speak zombie

1080pCat commented:
> > The plague claws should deal less damage to humans, they're a bit strong currently. Maybe each hit advancing the disease?
> 
> Moved claw damage down to 26 from 35 (4 hits will crit, down from 3) and made hits have a 40% chance to advance their respective disease
> 
> > Also the wizard can understand zombies, but cant /speak/ zombie, leading to iffy one way communication charades

I haven't seen the new claws in practice yet, so i'm just speaking from vibe on this
26 still feels a *tad* high. while its 14 less than necro stone, the claws are undropable, and the zombie passively regens(and revives)


With that being said, I'd hazard against lowering it without seeing it in action with the new numbers though.

Vilshen commented:
>and makes wiz/zombies virus immune so they wont be affected by their own fun

I've seen a zombie get infected by Grut Gut _and_ actually get thrown around by vomiting, so that doesn't work properly.

lewcc commented on code/datums/components/zombie_regen.dm:
```diff
@@ -16,10 +16,17 @@
 	var/turf/current_turf = get_turf(zomboid)
 	var/healing_factor = max(1, 6 * (1 - current_turf.get_lumcount()))
+	if(HAS_TRAIT(zomboid, TRAIT_PLAGUE_ZOMBIE))
+		healing_factor = healing_factor * 4 // plague zombies have more health, and should heal faster
```
```suggestion
		healing_factor *= 4 // plague zombies have more health, and should heal faster
```

lewcc commented on code/datums/components/zombie_regen.dm:
```diff
@@ -16,10 +16,17 @@
+	for(var/obj/item/organ/external/e in zomboid.bodyparts)
+		if(e.status & ORGAN_BURNT) // lets slowly remove any crit burns
+			if(prob(20))
+				e.status -= ORGAN_BURNT
```
This isn't generally how we remove one flag on things, I think you can look around for an &= to find how we tend to do it.

Also, please uppercase the `e` variable in this proc.

EmeraldCandy replied:
fixed

lewcc commented on code/datums/diseases/wizard_diseases.dm:
```diff
@@ -0,0 +1,233 @@
+			if(prob(2) && affected_mob.nutrition >= 100)
+				affected_mob.adjustBruteLoss(5)
+				affected_mob.vomit(6,0,TRUE,7,0)
+				affected_mob.visible_message("<span class='danger'>[affected_mob] vomits with such force that [p_theyre(FALSE)] sent flying backwards!</span>", "<span class='userdanger'>You vomit a torrent of magic bile so forcefully, that you are sent flying!</span>", "<span class='warning'>You hear someone vomit profusely.</span>")
```
```suggestion
				affected_mob.visible_message(
					"<span class='danger'>[affected_mob] vomits with such force that [affected_mob.p_theyre(FALSE)] sent flying backwards!</span>", 
					"<span class='userdanger'>You vomit a torrent of magic bile so forcefully, that you are sent flying!</span>", 
					"<span class='warning'>You hear someone vomit profusely.</span>"
				)
```

lewcc commented on code/datums/diseases/wizard_diseases.dm:
```diff
@@ -0,0 +1,233 @@
+			if(prob(3))
+				affected_mob.emote("sniff")
+			if(prob(3))
+				to_chat(affected_mob, "<span class='danger'>Your nose and throat feels uncomfortably dry.</span>")
```
```suggestion
				to_chat(affected_mob, "<span class='danger'>Your nose and throat feel uncomfortably dry.</span>")
```

lewcc commented on code/datums/diseases/wizard_diseases.dm:
```diff
@@ -0,0 +1,233 @@
+			if(prob(3))
+				affected_mob.emote("sniff")
+			if(prob(3))
+				to_chat(affected_mob, "<span class='danger'>Your nose and throat feels uncomfortably dry.</span>")
```
```suggestion
				to_chat(affected_mob, "<span class='danger'>Your nose and throat feel uncomfortably dry.</span>")
```

lewcc commented on code/datums/diseases/wizard_diseases.dm:
```diff
@@ -0,0 +1,233 @@
+			if(prob(2))
+				affected_mob.emote("cough")
+			if(prob(2))
+				to_chat(affected_mob, "<span class='danger'>Your nose feels like its going to burn right off your face!</span>")
```
```suggestion
				to_chat(affected_mob, "<span class='danger'>Your nose feels like it's going to burn right off your face!</span>")
```

lewcc commented on code/datums/diseases/wizard_diseases.dm:
```diff
@@ -0,0 +1,233 @@
+				T.add_vomit_floor(FALSE, FALSE, /obj/effect/decal/cleanable/vomit/spacematter)
+
+/obj/effect/decal/cleanable/vomit/spacematter
+	name = "Spacematter Vomit"
```
Lowercase it so it doesn't get treated as a proper noun
```suggestion
	name = "spacematter vomit"
```

lewcc commented on code/datums/diseases/wizard_diseases.dm:
```diff
@@ -0,0 +1,233 @@
+	milla.invoke_async(src)
+	if(world.time >= time_to_live) // dont want to remove the spot, but stop removing atmos
+		STOP_PROCESSING(SSprocessing, src)
+		src.visible_message("<span class = 'danger'>The spacematter vomit evaporates into nothingness!</span>")
```
```suggestion
		src.visible_message("<span class='danger'>[src] evaporates into nothingness!</span>")
```

lewcc commented on code/datums/diseases/zombie_virus.dm:
```diff
@@ -126,7 +135,14 @@
 	if(affected_mob.mind && !affected_mob.mind.has_antag_datum(/datum/antagonist/zombie))
-		affected_mob.mind.add_antag_datum(/datum/antagonist/zombie)
+		if(HAS_TRAIT(affected_mob, TRAIT_PLAGUE_ZOMBIE))
+			var/datum/antagonist/zombie/plague = new/datum/antagonist/zombie(plague_disease)
```
```suggestion
			var/datum/antagonist/zombie/plague = new /datum/antagonist/zombie(plague_disease)
```

lewcc commented on code/game/gamemodes/wizard/artefact.dm:
```diff
@@ -701,6 +700,10 @@ GLOBAL_LIST_EMPTY(multiverse)
 		return
 
+	if(victim.mind.has_antag_datum(/datum/antagonist/mindslave/necromancy/plague_zombie))
+		to_chat(necromancer, "<span class='warning'>This one is already under another artefact's influence!</span>")
```
```suggestion
		to_chat(necromancer, "<span class='warning'>This one is already under another artifact's influence!</span>")
```

lewcc commented on code/game/gamemodes/wizard/artefact.dm:
```diff
@@ -842,6 +845,99 @@ GLOBAL_LIST_EMPTY(multiverse)
+//////////////////////// plague Talisman //////////////////////////////
+
+/obj/item/plague_talisman
+	name = "plague Talisman"
```
```suggestion
	name = "\improper Plague Talisman"
```

lewcc commented on code/game/gamemodes/wizard/artefact.dm:
```diff
@@ -842,6 +845,99 @@ GLOBAL_LIST_EMPTY(multiverse)
+	for(var/datum/disease/zombie/wizard/zomb in victim.viruses)
+		zomb.stage = 8 // immediate zombie!
+
+	// Wiz and minions should contract their own diseases
```
misleading comment?

lewcc commented on code/game/gamemodes/wizard/artefact.dm:
```diff
@@ -842,6 +845,99 @@ GLOBAL_LIST_EMPTY(multiverse)
+	addtimer(CALLBACK(src, PROC_REF(finish_convert), victim, necromancer, chosen_plague), 5 SECONDS)
+
+/obj/item/plague_talisman/proc/finish_convert(mob/living/carbon/human/victim, mob/living/carbon/human/necromancer, datum/disease/chosen_plague)
+	var/greet_text = "<span class='userdanger'>You have been raised into undeath by <b>[necromancer.real_name]</b>!\n[necromancer.p_theyre(TRUE)] your master now, assist them at all costs, for you are now above death!<br> \
```
```suggestion
	var/greet_text = "<span class='userdanger'>You have been raised into undeath by <b>[necromancer.real_name]</b>!<br>
	[necromancer.p_theyre(TRUE)] your master now, assist [necromancer.p_them()]at all costs, for you are now above death!<br> \
```

lewcc commented on code/game/objects/items/weapons/storage/belt.dm:
```diff
@@ -708,6 +708,22 @@
 	update_icon()
 
+/obj/item/storage/belt/plague_talisman
+	name = "Runic Belt"
```
```suggestion
	name = "\improper Runic Belt"
```

lewcc commented on code/datums/diseases/wizard_diseases.dm:
```diff
@@ -0,0 +1,237 @@
+				to_chat(affected_mob, "<span class='danger'>A deep bubbling resounds through your chest.</span>")
+		if(5)
+			if(prob(2))
+				affected_mob.custom_emote(EMOTE_VISIBLE, "belches loudly!")
```
Should probably be `EMOTE_AUDIBLE`

lewcc commented on code/datums/diseases/wizard_diseases.dm:
```diff
@@ -0,0 +1,237 @@
+				to_chat(affected_mob, "<span class='danger'>Your abdomen gurgles and bubbles with a fierce intensity!</span>")
+			if(prob(2) && affected_mob.nutrition >= 100)
+				affected_mob.adjustBruteLoss(5)
+				affected_mob.vomit(6,0,TRUE,7,0)
```
```suggestion
				affected_mob.vomit(6, 0, TRUE, 7, 0)
```

lewcc commented on code/game/gamemodes/wizard/artefact.dm:
```diff
@@ -842,6 +845,103 @@ GLOBAL_LIST_EMPTY(multiverse)
+
+/obj/item/plague_talisman/proc/finish_convert(mob/living/carbon/human/victim, mob/living/carbon/human/necromancer, datum/disease/chosen_plague)
+	var/greet_text = "<span class='userdanger'>You have been raised into undeath by <b>[necromancer.real_name]</b>!<br> \
+	[necromancer.p_theyre(TRUE)] your master now, assist [necromancer.p_them()]at all costs, for you are now above death!<br> \
```
```suggestion
	[necromancer.p_theyre(TRUE)] your master now, assist [necromancer.p_them()] at all costs, for you are now above death!<br> \
```

lewcc commented on code/game/gamemodes/wizard/artefact.dm:
```diff
@@ -842,6 +845,103 @@ GLOBAL_LIST_EMPTY(multiverse)
+	// Cant very well have your new minions dead for so long. Make em stronk!
+	victim.maxHealth = 200
+	victim.health = 200
+	victim.rejuvenate()
```
This...won't cure their diseases, right?

EmeraldCandy replied:
nope

lewcc commented on code/game/gamemodes/wizard/artefact.dm:
```diff
@@ -842,6 +845,103 @@ GLOBAL_LIST_EMPTY(multiverse)
+		/datum/disease/grut_gut,
+		/datum/disease/wand_rot,
+		/datum/disease/mystic_malaise
+		)
```
```suggestion
	)
```

lewcc commented on code/modules/antagonists/zombie/datum_zombie.dm:
```diff
@@ -91,4 +109,15 @@ RESTRICT_TYPE(/datum/antagonist/zombie)
+		return
 	add_antag_objective(/datum/objective/zombie)
+
+/datum/antagonist/zombie/get_antag_objectives()
+	if(HAS_TRAIT(owner.current, TRAIT_PLAGUE_ZOMBIE))
+		return
+
+/datum/antagonist/zombie/add_antag_objective()
+	if(HAS_TRAIT(owner.current, TRAIT_PLAGUE_ZOMBIE))
+		return
```
These functions aren't doing anything, did you mean to invoke the parent?

EmeraldCandy replied:
invoked parents now

lewcc reviewed (commented):
Looking pretty good! Just a few nits at this point

lewcc reviewed (approved):
Code looks good to me, thanks for sticking with it!
